e2e/tests/protos.crud-all-fields.spec.ts

```diff
@@ -0,0 +1,142 @@
+
+import { API_PROTOS } from '@/config/constant';
+import type { APISIXType } from '@/types/schema/apisix';
+
+const protoContent = `syntax = "proto3";
+package test;
+
+message TestMessage {
+  string name = 1;
+  int32 age = 2;
+  string email = 3;
+}`;
+
+let createdProtoId: string;
+
+test.describe('CRUD proto with all fields', () => {
+  test.describe.configure({ mode: 'serial' });
+
+  test.afterAll(async () => {
+    // cleanup: delete the proto
+    if (createdProtoId) {
+      await e2eReq.delete(`${API_PROTOS}/${createdProtoId}`).catch(() => {
+        // ignore error if proto doesn't exist
+      });
+    }
+  });
+
+  test('should create a proto with all fields', async ({ page }) => {
+    await test.step('navigate to add proto page', async () => {
+      await protosPom.toAdd(page);
+      await protosPom.isAddPage(page);
+    });
+
+    await test.step('fill in all fields', async () => {
+      // Fill Content (ID is auto-generated, proto only has content field)
+      await page.getByLabel('Content').fill(protoContent);
+    });
+
+    await test.step('submit the form', async () => {
+      await page.getByRole('button', { name: 'Add', exact: true }).click();
+
+      // Should redirect to list page after successful creation
+      await protosPom.isIndexPage(page);
+    });
+
+    await test.step('verify proto was created via API', async () => {
+      // Get the list of protos to find the created one
+      const protos = await e2eReq
+        .get<unknown, APISIXType['RespProtoList']>(API_PROTOS)
+        .then((v) => v.data);
+
+      // Find the proto with our content (search for exact package name)
+      const createdProto = protos.list.find((p) =>
+        p.value.content?.includes('package test;')
+      );
+      expect(createdProto).toBeDefined();
+      expect(createdProto?.value.id).toBeDefined();
+      // eslint-disable-next-line playwright/no-conditional-in-test
+      createdProtoId = createdProto?.value.id || '';
+
+      // Verify content matches
+      expect(createdProto?.value.content).toBe(protoContent);
+    });
+  });
+
+  test('should read/view the proto details', async () => {
+    await test.step('verify proto can be retrieved via API', async () => {
+      const proto = await e2eReq
+        .get<unknown, APISIXType['RespProtoDetail']>(
+          `${API_PROTOS}/${createdProtoId}`
+        )
+        .then((v) => v.data);
+
+      expect(proto.value?.id).toBe(createdProtoId);
+      expect(proto.value?.content).toBe(protoContent);
+      expect(proto.value?.create_time).toBeDefined();
+      expect(proto.value?.update_time).toBeDefined();
+    });
+  });
+
+  test('should update the proto with new values', async () => {
+    const updatedContent = `syntax = "proto3";
+package test_updated;
+
+message UpdatedTestMessage {
+  string updated_name = 1;
+  int32 updated_age = 2;
+  string email = 3;
+  bool is_active = 4;
+}`;
+
+    await test.step('update proto via API', async () => {
```

Comment:
These operations should all be performed through the UI, which is also the purpose of our end-to-end testing.